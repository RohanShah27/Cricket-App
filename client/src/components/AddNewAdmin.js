import React, { Component } from "react";
import { addAdmin, sendPassword } from "../actions/User";

import { connect } from "react-redux";
import "../styles/AddNewAdmin.css";
export class AddNewAdmin extends Component {
    constructor(props) {
        super(props);
        //To allow the function this.onaddAdminClick to send data to the actions page -yash
        this.onaddAdminClick = this.onaddAdminClick.bind(this);
    }
    //The values that are going to be used in this component -yash
    state = {
        email: "",
        password: "123456"
    };

    OnChange = event => {
        //To change the text fields to accept input -yash
        this.setState({ [event.target.name]: event.target.value });
    };
    componentDidMount() {
        //If Token not present return to landing screen -yash
        if (!localStorage.getItem("token")) {
            this.props.history.push("/");
        }
    }
    //(e) used to prevent data being shown on the searchbar -yash
    onaddAdminClick(e) {
        e.preventDefault();
        //storing values email and password in an object user -yash
        let user = {
            email: this.state.email,
            password: this.state.password
        };
        //calling addAdmin function from actions -yash
        this.props.addAdmin(user)
        //calling sendpassword function from actions -yash
        this.props.sendPassword(user);
        //set current state of the values email and password -yash
        this.setState({
            email: "",
            password: "123456"
        });
    }

    render() {
        return (
            // /start of main div -yash
            <div>
                {/* start of form -yash */}
                <form id="addnewadminform">
                    {/* start of fieldset -yash */}
                    <fieldset>
                        {/* start and end of header -yash */}
                        <h1 className="adminheader">Add New Admin</h1>
                        {/* start of input tag email -yash */}
                        <input
                            className="adminnewinput"
                            type="text"
                            name="email"
                            placeholder="Enter email"
                            onChange={this.OnChange}
                            value={this.state.email}
                        />
                        {/* end of input tag email -yash */}
                        {/* display error from node  -yash */}
                        {this.props.error ? <><p>{this.props.error}</p></> : null}
                        {/* start of button add user -yash */}
                        <button
                            onChange={this.onChange}
                            onClick={this.onaddAdminClick}
                            className="admin-new-button"
                        >
                            Add User
            </button>
                        {/* end of button add user -yash */}
                    </fieldset>
                    {/* end of fieldset -yash */}
                </form>
                {/* end of form -yash */}
            </div>
            // end of div -yash
        );
    }
}

const mapStateToProps = state => ({
    //value entered in react is passed throught user reducer -yash
    users: state.userReducer.users,
    error: state.userReducer.error
});

export default connect(
    mapStateToProps,
    //functions (addAdmin) for creation of new admin and (sendPassword) send password via email on nodemailer-yash
    { addAdmin, sendPassword }
)(AddNewAdmin);
