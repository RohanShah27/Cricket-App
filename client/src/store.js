import { createStore, applyMiddleware, compose } from "redux";

import thunk from "redux-thunk";
import rootReducer from "./reducers";

const initialState = {};
//think will give all redducer actions to the middleware object
const middleware = [thunk];
// will give access to the extension
//   //We do this to check if chrome has the redux extension or not if not then it installs the devtoolextension in chrome
const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
      })
    : compose;
const enhancer = composeEnhancers(applyMiddleware(...middleware));
const store = createStore(rootReducer, initialState, enhancer);
export default store;
