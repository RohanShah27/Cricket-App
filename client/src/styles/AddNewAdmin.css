/* for button tag  -yash*/
.admin-new-button {
  width: 150px;
  background: white;
  color: #f39c12;
  font-size: 18px;
  border: 1px solid #c2c2c2;
  box-sizing: border-box;
  border-radius: 8px;
  cursor: pointer;
  padding: 10px;
  outline: 0;
  margin-bottom: 20px;
  box-shadow: 0px 4px 6px 0px #c2c2c2;
  margin-top: 20px;
}
/* for mobile query -yash */
@media screen and (max-width: 700px) {
  .col-25,
  .col-75 {
    width: 100%;
    margin-top: 0;
  }
}
/* for mobiile query -yash */
@media screen and (max-width: 500px) {
  .col-25,
  .col-75 {
    width: 100%;
    margin-top: 0;
  }
}
/* hover on button function -yash */
.admin-new-button:hover {
  background: #c2c2c2;
  color: white;
}
/* each placeholder in the component -yash */
::placeholder {
  color: #f39c12;
}
/* header function -yash */
.adminnewheader {
  font-size: 15px;
  text-transform: uppercase;
  color: #2c3e50;
  margin-bottom: 40px;
}
/* input fields -yash */
.adminnewinput {
  padding: 15px;
  border: 1px solid #c2c2c2;
  margin-top: 30px;
  border-radius: 8px;
  margin-bottom: 20px;
  width: 100%;
  box-sizing: border-box;
  color: #218380;
  font-size: 13px;
  outline: 0;
}
/* fieldset in addnewadminform -yash */
#addnewadminform fieldset {
  border: 1px solid #c2c2c2;
  border-radius: 8px;
  padding: 50px 50px;
  box-sizing: border-box;
  height: 320px;
  width: 80%;
  margin: 0 10%;
  position: relative;
}
/* form tag addnewadmin form -yash */
#addnewadminform {
  width: 500px;
  height: auto;
  margin: 50px auto;
  text-align: center;
  /* padding-bottom: 100px; */
  /* position: relative; */
}
