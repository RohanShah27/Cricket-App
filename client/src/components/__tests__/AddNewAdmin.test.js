import React from "react";
import { mount } from "enzyme";
import AddNewAdmin from "../AddNewAdmin";
import { Provider } from "react-redux";
import store from "../../store";

// const AddNewAdmin = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <AddNewAdmin addnewadmin={AddNewAdmin} history={[]} />
  </Provider>
);

describe("Test AddNewAdmin Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have exactly one input fields", () => {
    expect(wrapper.find("input").length).toBe(1);
  });

  it("should have exactly one button", () => {
    expect(wrapper.find("button").length).toBe(1);
  });

  it("should have exactly add user text on Button", () => {
    expect(wrapper.find("button").text()).toEqual("Add User");
  });

  it("should have Email placeholder on email input field", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .props().placeholder
    ).toBe("Enter email");
  });
  it("should simulate add admin button", () => {
    expect(
      wrapper
        .find("button")
        .at(1)
        .simulate("click")
    );
  });
});
