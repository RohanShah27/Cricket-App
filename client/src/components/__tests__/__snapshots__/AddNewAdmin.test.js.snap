// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddNewAdmin Component should render the component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(AddNewAdmin)
    addnewadmin={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(AddNewAdmin)",
        "type": [Function],
      }
    }
    history={
      Array [
        "/",
      ]
    }
  >
    <AddNewAdmin
      addAdmin={[Function]}
      addnewadmin={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AddNewAdmin)",
          "type": [Function],
        }
      }
      error=""
      history={
        Array [
          "/",
        ]
      }
      sendPassword={[Function]}
      users={Array []}
    >
      <div>
        <form
          id="addnewadminform"
        >
          <fieldset>
            <h1
              className="adminheader"
            >
              Add New Admin
            </h1>
            <input
              className="adminnewinput"
              name="email"
              onChange={[Function]}
              placeholder="Enter email"
              type="text"
              value=""
            />
            <button
              className="admin-new-button"
              onClick={[Function]}
            >
              Add User
            </button>
          </fieldset>
        </form>
      </div>
    </AddNewAdmin>
  </Connect(AddNewAdmin)>
</Provider>
`;
